packages/react-native/src/helpers.ts

```diff
@@ -0,0 +1,54 @@
+import { useState, useEffect, useMemo } from "react";
+import { Animated, Easing } from "react-native";
+import { Themes } from "@pubnub/common-chat-components";
+
+export interface UseStyleProps<T> {
+  theme: Themes | "";
+  createDefaultStyle: (theme: Themes | "") => T;
+  customStyle?: T;
+}
+
+export const useStyle = <T>(props: UseStyleProps<T>): T => {
+  const { theme, createDefaultStyle, customStyle } = props;
+  const [style, setStyle] = useState(createDefaultStyle(theme));
+
+  useEffect(() => {
+    setStyle(Object.assign({}, createDefaultStyle(theme), customStyle));
+  }, [createDefaultStyle, setStyle, customStyle, theme]);
+
+  return style;
+};
+
+export const useRotation = (shouldRun = true): Animated.AnimatedInterpolation => {
+  const [rotationValue] = useState(new Animated.Value(0));
+  const animation = useMemo(
+    () =>
+      Animated.loop(
+        Animated.timing(rotationValue, {
+          toValue: 1,
+          duration: 650,
```

Comment:
Also I'd like to see that we do have animations defined in one place e.g. LINEAR_ANIM, CASCADING_ANIM so that we do not create many different animations with magic numbers